src/main/java/net/fabricmc/loader/entrypoint/EntrypointTransformer.java

```diff
@@ -42,7 +42,7 @@ public EntrypointTransformer(Function<EntrypointTransformer, List<EntrypointPatc
 	}
 
 	ClassNode loadClass(FabricLauncher launcher, String className) throws IOException {
-		byte[] data = patchedClasses.containsKey(className) ? patchedClasses.get(className) : launcher.getClassByteArray(className, false);
```

Comment:
Why? I haven't looked at this in detail, but this looks at first glance like it might cause an infinite loop. I.e. EntrypointTransformer asks to run transformers which runs the EntrypointTransformer

Reply:
It's not. The EntrypointTransformer is not an actual tranformer, i.e. it's never called in response to class loading. It's merely called during bootstrap, loads the bytecode, applies its patch and then stores the result in a map.
The real transformer merely queries that map, if it has an entry (and at this point it won't), it returns that and otherwise it just delegates further.